controllers/user/UserProfile.js

```diff
@@ -47,38 +46,74 @@ class UserProfile {
 
   /**
    *
-   * @param {request} request HTTP request
-   * @param {response} response HTTP response
-   * @param {response} next HTTP next response
-   * @return {object} return response to user
+   *
+   * @static
+   * @param {request} req
+   * @param {response} res
+   * @param {function} next
+   * @returns {object} returns profile for a specific user
+   * @memberof UserProfile
    */
-  static getProfile(request, response, next) {
-    const { userId } = request.authData;
-    return User
-      .findById(userId) // collect user Id from token
-      .then((user) => {
-        if (!user) {
-          return response.status(404)
-            .json({
-              message: 'User is not found'
-            });
-        }
-        return response.status(200)
-          .json({
-            profile: {
-              email: user.email,
-              username: user.username,
-              firstName: user.firstName,
-              lastName: user.lastName,
-              bio: user.bio,
-              imageUrl: user.imageUrl,
-              createdAt: user.createdAt,
-              updatedAt: user.updatedAt
-            },
-            message: 'Successful!',
-          });
-      })
-      .catch(error => next(error));
+  static async getProfileById(req, res, next) {
+    const id = req.params.id || req.authData.userId;
+    try {
+      const user = await User.findById(id);
+      HttpError.throw404ErrorIfNull(user, 'User not found');
+      const profile = UserHelper.profileListResponse(user);
+      res.status(200).json({
+        status: 'success',
+        message: 'Successfully retrieved user profile!',
+        profile
+      });
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {request} req
+   * @param {response} res
+   * @param {function} next
+   * @returns {object} return a json response
+   * @memberof UserProfile
+   */
+  static async getAllProfile(req, res, next) {
```

Comment:
@Alameen688 There is an error with the pagination you implemented. In your `User.findAll` you are excluding the currently logged in user, but `User.count()` will include that user in the count. This makes the pagination inaccurate. View screenshot
<img width="1324" alt="screen shot 2018-10-11 at 4 17 17 pm" src="https://user-images.githubusercontent.com/15909231/46814687-2b85fd00-cd71-11e8-8f20-1de192e368bf.png">

From the screenshot, if the total number of users is 5 when page=3 and size=2, the user should get one result not an empty array.

Reply:
Thanks for spotting that. @macphilips. Corrected!